lib/models/users-schema.js

```diff
@@ -1,3 +1,79 @@
+// 'use strict';
+
+// // == EXTERNAL RESOURCES ===============================================
+
+// const mongoose = require('mongoose');
+// const bcrypt = require('bcrypt');
+// const jwt = require('jsonwebtoken');
+
+// // == DEFINE THE USER SCHEMA =============================================
+
+// // Mongoos schema for users documents 
+// const users = mongoose.Schema({
+//   username: { type: String, required: true, unique: true },
+//   password: { type: String, required: true },
+//   email: { type: String },
+//   role: {
+//     type: String,
+//     required: true,
+//     default: 'user',
+//     enum: ['admin', 'editor', 'user']
+//   }
+// });
+
+// /**
+//  * @param  {} 'password'
+//  * before save to users check if password was modified. 
+//  * Hash password with bcrypt if modified. 
+//  */
+
+// users.pre('save', async function() {
+//   if (this.isModified('password')) {
+//     this.password = await bcrypt.hash(this.password, 10);
+//   }
+// });
+
+// /**
+//  * Takes user's creds, looks for creds.username match in database, 
+//  * if found calls comparePassword() 
+//  * @param  {object} creds
+//  * @param  {creds.username}
+//  * @param  {creds.password}
+//  * @returns true if auth is successful, false if not 
+//  */
+// users.statics.authenticate = async function(creds) {
+//   let query = { username: creds.username };
+
+//   let user = await this.findOne(query);
+//   let isValid = await user.comparePasswords(creds.password);
+// };
+
+// /**
+//  * Compares password passed into the funciton with this user password.
+//  * If passwords are equel - returns user object, if not - null
+//  * @param  {} password
+//  * @returns {object} || NULL
+//  */
+// users.methods.comparePassword = async function(password) {
+//   let hashedPassword = this.password;
+//   let valid = await bcrypt.compare(password, hashedPassword);
+//   return valid;
```

Comment:
The return value is either an `{object} || NULL `.  Inside the function if the passwords don't match, will it return null?